api/envoy/config/endpoint/v3/endpoint_components.proto

```diff
@@ -44,6 +44,13 @@ message Endpoint {
     // to a non-empty value allows overriding the cluster level configuration for a specific
     // endpoint.
     string hostname = 2;
+
+    // Optional alternative health check host address.
+    //
+    // .. attention::
+    //
+    //   The form of the health check host address is expected to be a direct IP address.
```

Comment:
Please add a test for the case where this is not a direct IP address and make sure the failure message is actionable for the user.